src/one_of.rs

```diff
@@ -34,14 +35,14 @@ pub struct OneOf<E: TypeSet> {
     _pd: PhantomData<E>,
 }
 
-fn _send_sync_error_assert() {
-    use std::io;
-
+#[cfg(test)]
+#[test]
+fn send_sync_error_assert() {
     fn is_send<T: Send>(_: &T) {}
     fn is_sync<T: Sync>(_: &T) {}
     fn is_error<T: Error>(_: &T) {}
 
-    let o: OneOf<(io::Error,)> = OneOf::new(io::Error::new(io::ErrorKind::Other, "yooo"));
+    let o: OneOf<(core::fmt::Error,)> = OneOf::new(core::fmt::Error);
```

Comment:
`io::Error` is not `Clone`, can we introduce a custom error type that also is not `Clone` to make sure nothing material changes about the test?